### PR TITLE
Revert default role assignment for New-AzureRmADServicePrincipal

### PR DESCRIPTION
**Pull request by cormacpayne**
## Description

- Revert the change to `New-AzureRmADServicePrincipal` that gave "Contributor" permissions to the current subscription by default if `Role` and `Scope` parameters weren't provided. The behavior now goes back to before the 6.0.0 release where no permissions are assigned to the service principal if `Role` and `Scope` aren't provided, but use the values provided (or default values) if one (or both) of the parameters are provided.
- Update the examples for `New-AzureRmADServicePrincipal` to reflect these changes, as well as update the descriptions for the cmdlet and the affected parameters

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by cormacpayne** [comment]
On-demand: https://azuresdkci.westus2.cloudapp.azure.com/job/powershell-demand/869/
Sign job: https://azuresdkci.westus2.cloudapp.azure.com/job/ps-sign/644/

File: src/ResourceManager/Resources/ChangeLog.md [hunk 1]
```diff
@@ -18,6 +18,11 @@
         - Additional information about change #1
 -->
 ## Current Release
+* Revert change to `New-AzureRmADServicePrincipal` that gave service principals "Contributor" permissions over the current subscription if no values were provided for the `Role` or `Scope` parameters
+    - If no values are provided for `Role` or `Scope`, the service principal is created with no permissions
+    - If a `Role` is provided, but no `Scope`, the service principal is created with the specified `Role` permissions over the current subscription
+    - If a `Scope` is provided, but no `Contributor`, the service principal is created with `Contributor` permissions over the specified `Scope`
```
Comment (darshanhs90) [inline src/ResourceManager/Resources/ChangeLog.md]: >Contributor [](start = 40, length = 11)

role

**Comment by darshanhs90** [comment]
>                 Role = "Contributor";

why are we defaulting to contributor role,when a role isnt provided?

---
Refers to: src/ResourceManager/Resources/Commands.Resources/ActiveDirectory/NewAzureADServicePrincipalCommand.cs:281 in 0acb1cf. [](commit_id = 0acb1cf4868218f84c80a747198add49732d1619, deletion_comment = False)

**Comment by markcowl** [comment]
@darshanhs90 Because that is our best guess at what the user would want in this case - since no previous versions of the cmdlet have role or scope as a parameter, there is no chance of an existing script hitting unexpected behavior in this case, as they have to provide at least 'Scope'

**Comment by twitchax** [comment]
@darshanhs90, the new behavior is meant to match the Azure CLI behavior, so the default Role is Contributor, as it is in CLI.

**Comment by cormacpayne** [comment]
Sign job: https://azuresdkci.westus2.cloudapp.azure.com/job/ps-sign/659/

**Comment by cormacpayne** [comment]
@darshanhs90 @markcowl comments addressed and tests added

File: ChangeLog.md [hunk 2]
```diff
@@ -1,4 +1,12 @@
-﻿## 6.1.0 - May 2018
+﻿## 6.1.1 - May 2018
+#### AzureRM.Resources
+* Revert change to `New-AzureRmADServicePrincipal` that gave service principals `Contributor` permissions over the current subscription if no values were provided for the `Role` or `Scope` parameters
+    - If no values are provided for `Role` or `Scope`, the service principal is created with no permissions
+    - If a `Role` is provided, but no `Scope`, the service principal is created with the specified `Role` permissions over the current subscription
+    - If a `Scope` is provided, but no `Scope`, the service principal is created with `Contributor` permissions over the specified `Scope`
```
Comment (darshanhs90) [inline ChangeLog.md]: Role?

File: src/ResourceManager/Resources/ChangeLog.md [hunk 3]
```diff
@@ -19,6 +19,13 @@
 -->
 ## Current Release
 
+## Version 6.0.1
+* Revert change to `New-AzureRmADServicePrincipal` that gave service principals `Contributor` permissions over the current subscription if no values were provided for the `Role` or `Scope` parameters
+    - If no values are provided for `Role` or `Scope`, the service principal is created with no permissions
+    - If a `Role` is provided, but no `Scope`, the service principal is created with the specified `Role` permissions over the current subscription
+    - If a `Scope` is provided, but no `Scope`, the service principal is created with `Contributor` permissions over the specified `Scope`
```
Comment (darshanhs90) [inline src/ResourceManager/Resources/ChangeLog.md]: Role

File: tools/AzureRM/AzureRM.psd1 [hunk 4]
```diff
@@ -159,67 +159,13 @@ PrivateData = @{
...
-	- Restore-AzureRmSqlDatabase
-
-AzureRM.TrafficManager
-* Update the parameters for ''Get-AzureRmTrafficManagerProfile'' so that -ResourceGroupName parameter is required when using -Name parameter.
+        ReleaseNotes = '6.1.1 - May 2018
+AzureRM.Resources
+* Revert change to `New-AzureRmADServicePrincipal` that gave service principals `Contributor` permissions over the current subscription if no values were provided for the `Role` or `Scope` parameters
+    - If no values are provided for `Role` or `Scope`, the service principal is created with no permissions
+    - If a `Role` is provided, but no `Scope`, the service principal is created with the specified `Role` permissions over the current subscription
+    - If a `Scope` is provided, but no `Scope`, the service principal is created with `Contributor` permissions over the specified `Scope`
```
Comment (darshanhs90) [inline tools/AzureRM/AzureRM.psd1]: Role

File: src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/ActiveDirectoryTests.ps1 [hunk 5]
```diff
@@ -588,12 +588,78 @@ function Test-NewADServicePrincipalWithoutApp
     Assert-Throws { Remove-AzureRmADServicePrincipal -ObjectId $servicePrincipal.Id -Force}
 }
 
+<#
+.SYNOPSIS
+Tests creating a service principal with reader permissions
+#>
+function Test-NewADServicePrincipalWithReaderRole
+{
+	# Setup
```
Comment (darshanhs90) [inline src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/ActiveDirectoryTests.ps1]: can we add another test where in we dont specify any role or scope and make sure that no roleassignment change happens before and after running the command

Comment (cormacpayne) [reply src/ResourceManager/Resources/Commands.Resources.Test/ScenarioTests/ActiveDirectoryTests.ps1]: @darshanhs90 add a check in `Test-NewADServicePrincipalWithoutApp` that no role assignment was created for the new service principal

**Comment by cormacpayne** [comment]
Sign job: https://azuresdkci.westus2.cloudapp.azure.com/job/ps-sign/668/

File: ChangeLog.md (hunk 2 above)
Comment (cormacpayne) [reply ChangeLog.md]: @darshanhs90 oops, thanks. Fixed this in all occurrences.

**Pull request merged**